server/aesdsocket.c

```diff
@@ -117,36 +248,242 @@ void print_closed_conn(struct sockaddr_storage client_addr)
 }
 
 /**
-* exit_wrapper
-* @brief Closes socket and used files before exiting
+* serve_client
+* @brief Serves a client from a socket connection
 * 
-* @param int    socket fd
-* @param int    log fd+
+* The server functionality is described in more detail at
+* the file header.
+*
 * @return void
 */
-void exit_wrapper(int sck, int file_fd)
+void *serve_client(void *thread_info)
 {
-    //After stopping to accept requests, the socket can be closed
-    if(close(sck) == -1)
+    struct thread_t *thread_info_parsed = (struct thread_t *) thread_info;
+
+    print_accepted_conn(thread_info_parsed->client_addr);
+
+    //Wait for data
+    int recv_ret;
+    int index = 0;
+    //Keep track of how many RECV_BUFF_LEN chunks "recv_data" has
+    int chunks = 1;
+    
+    //Start with a size, potentially increasing it if an entire packet cannot fit into it
+    char *recv_data = malloc(sizeof(char)*RECV_BUFF_LEN*chunks);
+    if(!recv_data)
     {
-        //Else, error occurred, print it to syslog and finish program
-        syslog(LOG_ERR, "Could not close socket: %s", strerror(errno));
-        exit(1);
+        syslog(LOG_ERR, "Could not malloc: %s", strerror(errno));
+        goto exit_nofree; 
     }
-    //Close the file used to log all the data received
-    if(close(file_fd) == -1)
+
+    do
     {
-        //Else, error occurred, print it to syslog and finish program
-        syslog(LOG_ERR, "Could not close log file: %s", strerror(errno));
-        exit(1);
-    }
-    //Delete the file
-    if(remove(LOG_PATH) == -1)
+        recv_ret = recv(thread_info_parsed->socket_client, &recv_data[index], RECV_BUFF_LEN, 0);
+        if(recv_ret == -1)
+        {
+            syslog(LOG_ERR, "An error occurred reading from the socket: %s", strerror(errno));
+            goto exit_all;  
+        }
+        index += recv_ret;
+
+        if(index != 0)
+        {
+            //Check if the last value received is "\n"
+            if(recv_data[index - 1] == '\n')
```

Comment:
@jordicrosmompart see review notes in https://github.com/cu-ecen-aeld/assignments-3-and-later-jordicrosmompart/pull/3#discussion_r988334533 - I don't think this will handle the case where the received buffer includes two packets.